packages/artillery-plugin-memory-inspector/index.js

```diff
@@ -0,0 +1,63 @@
+const pidusage = require('pidusage');
+const debug = require('debug')('plugin:memory-inspector');
+
+function ArtilleryPluginMemoryInspector(script, events) {
+  this.script = script;
+  this.events = events;
+
+  const inspectorConfig =
+    script.config['memory-inspector'] ||
+    script.config.plugins['memory-inspector'];
+
+  async function memoryInspectorHandler(context, ee, next) {
+    if (typeof process.env.ARTILLERY_INTROSPECT_MEMORY !== 'undefined') {
+      //https://nodejs.org/api/process.html#processmemoryusage
+      const { rss, heapUsed, heapTotal, external } = process.memoryUsage();
+      ee.emit('histogram', 'artillery_internal.memory', rss);
+      ee.emit('histogram', 'artillery_internal.external', external);
+      ee.emit('histogram', 'artillery_internal.heap_used', heapUsed);
+      ee.emit('histogram', 'artillery_internal.heap_total', heapTotal);
+    }
+
+    for (let { pid, name } of inspectorConfig) {
+      if (!pid) {
+        debug(`No pid (${pid}) or name (${name}) found. Skipping!`);
+        continue;
+      }
+      let stats;
+      try {
+        stats = await pidusage(pid);
+
+        ee.emit('histogram', `${name || `process_${pid}`}.cpu`, stats.cpu);
+        ee.emit(
+          'histogram',
+          `${name || `process_${pid}`}.memory`,
+          stats.memory
+        );
+      } catch (error) {
+        debug(`Could not get usage stats for pid ${pid}.\n${error}`);
+        continue;
+      }
+    }
+
+    return next();
+  }
+
+  if (!script.config.processor) {
```

Comment:
Do you think it'd be better to move this fallback value closer to where we actually need it (at `script.config.processor.memoryInspectorHandler` assignment)?

Reply:
to be honest I wrote it like this because I looked around at other plugins for examples and it was similar there. Don't mind bringing it down though!